File: Napakalaki Java/NapakalakiGame/NumericBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author Germán
 */
public class NumericBadConsequence extends BadConsequence {
    
    private int nVisibleTreasures; //tesoros visibles que se pierden
    private int nHiddenTreasures; // tesoros ocultos que se pierden
    
    public NumericBadConsequence (String Text, int Levels, int nVisible, int nHidden) {
        super(Text, Levels);
        nVisibleTreasures = nVisible;
        nHiddenTreasures = nHidden;
    }
    
    public int getnVisibleTreasures () {
        return nVisibleTreasures;
    }
    
    public int getnHiddenTreasures () {
        return nHiddenTreasures;
    }
    
    @Override
    public String toString() {
        String cadena;
        cadena = "Levels = " + Integer.toString(super.levels) + 
                 " Visible treasures = " + Integer.toString(nVisibleTreasures) + 
                 " Hidden treasures = " + Integer.toString(nHiddenTreasures);
        
        return cadena;
    }
    
    //Preguntar si hay que dejarlo en la super clase
    @Override
    public boolean isEmpty() {
        return (nVisibleTreasures ==0 && nHiddenTreasures ==0);
    }
    
    @Override
    public void substractVisibleTreasure(Treasure t) {
        nVisibleTreasures--;
    } 
    
    @Override
    public void substractHiddenTreasure(Treasure t) {
        nHiddenTreasures--;
    }
    
    @Override
    public BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h) {
        NumericBadConsequence malrollo;
        malrollo = new NumericBadConsequence(super.text,levels,nVisibleTreasures,nHiddenTreasures);
        if(v.size()<nVisibleTreasures)
            malrollo.nVisibleTreasures = v.size();
        if(h.size()<nHiddenTreasures)
            malrollo.nHiddenTreasures = h.size();
        
        return malrollo;
    }
   
}

File: Napakalaki Ruby/NumericBadConsequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

class NumericBadConsequence < BadConsequence
  
  def initialize(texto, nivel, nVisible, nHidden)
    super(texto,nivel)
    
    @n_visible_treasures = nVisible
    @n_hidden_treasures = nHidden
  end
  
  def getNVisible
    return @n_visible_treasures
  end
  
  def getNHidden
    return @n_hidden_treasures
  end
 
  
  def to_s
    cadena = super
    cadena += "Niveles perdidos =  #{@levels}," + 
                      "Tesoros visibles perdidos = #{@n_visible_treasures}," + 
                      "Tesoros ocultos perdidos = #{@n_hidden_treasures} "
    return cadena                
  end
  
  def isEmpty
    vacio = false
    
    if(@n_visible_treasures ==0 and @n_hidden_treasures ==0)# or (@n_visible_treasures <-1 or @n_hidden_treasures <-1))
      vacio = true
    end
    
    return vacio
  end
  
  def substractVisibleTreasure(t)
    @n_visible_treasures = @n_visible_treasures -1
  end
  
  def substractHiddenTreasure(t)
    @n_hidden_treasures = @n_hidden_treasures -1
  end
  
  def adjustToFitTreasureList(v, h)
    
    tamav = @n_visible_treasures
    tamah = @n_hidden_treasures
    if(v.size()<@n_visible_treasures)
      tamav = v.size
    end
    if(h.size()<@n_hidden_treasures)
      tamah = h.size
    end
    malrollo = NumericBadConsequence.new(@text, @levels, tamav, tamah)
        
    return malrollo
  end
  
end

end
